mlflow/evaluation.py

```diff
@@ -0,0 +1,224 @@
+        for entrypoint in entrypoints.get_group_all("mlflow.model_evaluator"):
+            try:
+                self.register(entrypoint.name, entrypoint.load())
+            except (AttributeError, ImportError) as exc:
+                warnings.warn(
+                    'Failure attempting to register model evaluator for scheme "{}": {}'.format(
+                        entrypoint.name, str(exc)
+                    ),
+                    stacklevel=2,
+                )
+
+    def get_evaluator(self, evaluator_name):
+        """
+        Get an evaluator instance from the registry based on the name of evaluator
+        """
+        evaluator_cls = self._registry.get(evaluator_name)
+        if evaluator_cls is None:
+            raise MlflowException(
+                "Could not find a registered model evaluator for: {}. "
+                "Currently registered evaluator names are: {}".format(
+                    evaluator_name, list(self._registry.keys())
+                )
+            )
+        return evaluator_cls()
+
+
+_model_evaluation_registry = ModelEvaluatorRegistry()
+_model_evaluation_registry.register_entrypoints()
+
+
+def evaluate(
+   model, model_type, dataset, run_id=None, evaluators=None, evaluator_config=None
+) -> Union[EvaluationResult, Dict[str, EvaluationResult]]:
+    """
+    :param model: A model supported by the specified `evaluator`, or a URI
+                  referring to such a model. The default evaluator supports the
+                  following:
+
+                  - A pyfunc model instance (an instance of class `PyFuncModel`)
+
+    :param model_type: A string describing the model type. The default evaluator
+                       supports "regressor" and "classifier" as model types.
+    :param dataset: An instance of `EvaluationDataset` containing features
+                    labels (optional) for model evaluation.
+    :param run_id: The ID of the MLflow Run to which to log results. If
+                   unspecified, behavior depends on the specified `evaluator`.
+                   When `run_id` is unspecified, the default evaluator logs
+                   results to the current active run, creating a new active run if
+                   one does not exist.
+    :param evaluators: The name of the evaluator to use for model evaluations, or
+                       a list of evaluator names. If unspecified, all evaluators
+                       capable  of evaluating the specified model on the specified
+                       dataset are used.
+    :param evaluator_config: A dictionary of additional configurations to supply
+                             to the evaluator. If multiple evaluators are
+                             specified, each configuration should be supplied as
+                             a nested dictionary whose key is the evaluator name.
+    :return: An `EvaluationResult` instance containing evaluation results.
```

Comment:
@dbczumar When we specify a list of evaluators, What shall we return ? In this PR, I return a dict of "evaluator_name" --> evaluation_result

Reply:
@WeichenXu123 Can we merge the evaluation results into a single object? Otherwise, if there are multiple evaluators registered, users won't know whether the return type is `dict` or `EvaluationResult` when `evaluators=None`. Returning `EvaluationResult` in both cases is better than always returning a dictionary because, usually, only the default evaluator will be used, so we shouldn't have to make users unpack a `dict`.